### PR TITLE
[telegram-web-app] add support for Bot API 7.10

### PR DESCRIPTION
**Pull request by sidorko**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

If changing an existing definition:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [September 6, 2024](https://core.telegram.org/bots/webapps#september-6-2024)
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/telegram-web-app/telegram-web-app-tests.ts
1. types/telegram-web-app/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 252f1dffb277ce67f1a418bb4a99da4dafe56fbb
</p>

File: types/telegram-web-app/index.d.ts [hunk 1]
```diff
@@ -96,11 +100,16 @@ interface WebApp {
      * header of the Web App in the Telegram interface.
      */
     BackButton: BackButton;
+    /**
+     * An object for controlling the secondary button,
+     * which is displayed at the bottom of the Mini App in the Telegram interface.
+     */
+    MainButton: BottomButton;
```
Comment (KnorpelSenf) [inline types/telegram-web-app/index.d.ts:107]: Please fix the docs. The main button is documented as secondary button and vice-versa now.

File: types/telegram-web-app/index.d.ts [hunk 2]
```diff
@@ -129,11 +138,16 @@ interface WebApp {
      */
     // string & {} prevents this from eagerly collapsing into just string
     setHeaderColor(color: "bg_color" | "secondary_bg_color" | (string & {})): void;
+    /**
+     * A method that sets the app's bottom bar color in the #RRGGBB format.
+     * You can also use the keywords bg_color, secondary_bg_color and bottom_bar_bg_color.
+     */
+    setBackgroundColor(color: "bg_color" | "secondary_bg_color" | "bottom_bar_bg_color" | (string & {})): void;
```
Comment (KnorpelSenf) [inline types/telegram-web-app/index.d.ts:145]: Same thing here for background color and bottom bar color

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
Types LGTM, just a few bad docs:

**Comment by KnorpelSenf** [comment]
@sidorko just retry, I don't think the failing CI is related to your changes

**Review by KnorpelSenf — APPROVED** [review]
LGTM, thanks!

**Comment by KnorpelSenf** [comment]
Ready to merge

**Comment by KnorpelSenf** [comment]
lol

**Comment by KnorpelSenf** [comment]
Ready to merge

**Pull request merged**